pandas/core/groupby/ops.py

```diff
@@ -462,6 +469,11 @@ def _cython_op_ndim_compat(
             # otherwise we have OHLC
             return res.T
 
+        if self.kind == "transform" and self.grouper.has_dropped_na:
+            mask = comp_ids == -1
+            # make mask 2d
```

Comment:
can this be rolled into the mask-reshaping done above?

Reply:
This block turned out to not be necessary.